client/src/pages/Userprofile/UserWeatherSection.js

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useState } from 'react';
+import { useGeolocation } from '../UserLocation/useGeolocation';
+
+export default function UserWeatherSection() {
+  const date = new Date();
+  const months = [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December',
+  ];
+  const weekdays = [
+    'Sunday',
+    'Monday',
+    'Tuesday',
+    'Wednesday',
+    'Thursday',
+    'Friday',
+    'Saturday',
+  ];
+
+  const month = months[date.getMonth()];
+  const weekday = weekdays[date.getDay()];
+  const day = date.getDay();
+
+  const { data } = useGeolocation();
+
+  const [temperature, setTemperature] = useState(null);
+  const [weather, setWeather] = useState(null);
+  const [city, setCity] = useState(null);
+  const [country, setCountry] = useState(null);
+  const [windSpeed, setWindSpeed] = useState(null);
+  const [compassDirection, setCompassDirection] = useState('');
+
+  const fetchWeather = async () => {
+    if (!data || temperature) return;
+    const response = await fetch(
+      `http://api.openweathermap.org/data/2.5/weather?units=metric&lat=${data?.coords?.latitude}&lon=${data?.coords?.longitude}&appid=${API_KEY}`,
+    );
+    const json = await response.json();
+    console.log(json);
+    setTemperature(json.main.temp);
+    setWeather(json.weather[0].main);
+    setCity(json.name);
+    setCountry(json.sys.country);
+    setWindSpeed(json.wind.speed);
+    const points = Math.round(json.wind.deg / 22.5);
+    const directions = [
```

Comment:
Prefer the constants `directions`, `months`, `weekdays` to be at the top of the file outside of `UserWeatherSection` for readability

Reply:
I made the required changes.